src/libsyntax/errors/mod.rs

```diff
@@ -659,7 +659,7 @@ impl Level {
     fn color(self) -> term::color::Color {
         match self {
             Bug | Fatal | PhaseFatal | Error => term::color::BRIGHT_RED,
-            Warning => term::color::BRIGHT_YELLOW,
+            Warning => term::color::BRIGHT_MAGENTA,
```

Comment:
Could we detect the background colour? Yellow is traditional for warnings and magenta is pretty offensive on a black background